### PR TITLE
[ios, build] Fixup changelog and improve jazzy readme generation

### LINKED ISSUE #11443: Jazzy changelog should include all releases in semver minor series
The jazzy docs for each iOS release includes the changelog section for that release on the jazzy index page. For patch releases, this can be a bit sparse and the relatively faster release cadence can push significant changes out of view. 

We should include the full changelog for the entire release series — e.g., a patch release such as `3.7.6` should include the changelog for every release back to `3.7.0` (its semver minor parent), our next release series would include `4.0.x`, and so on.

Alternatively, we could include the _entire_ changelog, but collapse it to only show recent releases.

[This is the script that handles the jazzy readme generation.](https://github.com/mapbox/mapbox-gl-native/blob/5b3ebc4a2429c202f08c7468c300950de4844615/platform/ios/scripts/document.sh#L24-L28)

screen_shot_2018-03-13_at_12_52_06_pm

/cc @jmkiley @captainbarbosa @1ec5

### PR DESCRIPTION
**Pull request by friedbunny**
### Changelog fixes
Fixed a [one-level-too-high header](https://github.com/mapbox/mapbox-gl-native/pull/12704/files#diff-5b24c8d4438ee57ae98d7734b2e45557L25) in the 4.3.0 changelog (#12650), which broke the changelog `sed` incantation in our documentation generation script. This led to a truncated changelog in our jazzy readme, so we’ll need to update the docs in `ios-sdk` after merging this.

### Jazzy readme generation improvements
Replaced jazzy’s readme generation bash script with the more sophisticated templated JS script we’ve been using for GitHub release notes (#11997). This fixes #11443 by including changelogs from all releases in the current semver minor series.

mapbox maps sdk for ios reference 20180821

/cc @mapbox/maps-ios @1ec5

### CODE REVIEW

File: platform/ios/scripts/document.sh [hunk 1]
```diff
@@ -28,18 +28,18 @@ RELEASE_VERSION=$( echo ${SHORT_VERSION} | sed -e 's/^ios-v//' -e 's/-.*//' )
 
 rm -rf /tmp/mbgl
 mkdir -p /tmp/mbgl/
+
+echo "Generating readme and release notes"
 README=/tmp/mbgl/README.md
-cp platform/ios/docs/doc-README.md "${README}"
-# http://stackoverflow.com/a/4858011/4585461
-echo "## Changes in version ${RELEASE_VERSION}" >> "${README}"
-sed -n -e '/^## /{' -e ':a' -e 'n' -e '/^## /q' -e 'p' -e 'ba' -e '}' platform/ios/CHANGELOG.md >> "${README}"
```
Comment (1ec5) [inline platform/ios/scripts/document.sh]: One fewer opportunity for contributors to learn `sed`. 😢😂

File: platform/ios/scripts/release-notes-jazzy.md.ejs [hunk 2]
```diff
@@ -4,6 +4,17 @@ The Mapbox Maps SDK for iOS is an open-source framework for embedding interactiv
...
+For setup information, check out the [Mapbox Maps SDK for iOS homepage](https://www.mapbox.com/ios-sdk/). For detailed usage instructions, read “[First steps with the Mapbox Maps SDK for iOS](https://www.mapbox.com/help/first-steps-ios-sdk/)” and consult the [online examples](https://www.mapbox.com/ios-sdk/examples/).
 
 If you have any questions, please see [our help page](https://www.mapbox.com/help/). We welcome your [bug reports, feature requests, and contributions](https://github.com/mapbox/mapbox-gl-native/issues/).
+
+## Changes in version <%-CURRENTVERSION%>
+
+<%-CURRENTCHANGELOG-%>
+
+<% OTHERRELEASES.forEach(function(release) { %>
+## <%-release.version%>
```
Comment (1ec5) [inline platform/ios/scripts/release-notes-jazzy.md.ejs]: The version number on its own looks a little bare. How about prepending “Changes in version” like in the heading above?

Comment (friedbunny) [reply platform/ios/scripts/release-notes-jazzy.md.ejs]: I dunno, I’m not entirely happy either way — this seems redundant. 🤷‍♂️

mapbox maps sdk for ios reference 20180821

Comment (friedbunny) [reply platform/ios/scripts/release-notes-jazzy.md.ejs]: Perhaps a subtle `<hr>` somewhere?

Comment (1ec5) [reply platform/ios/scripts/release-notes-jazzy.md.ejs]: How about “Recent changes” as h2, “Changes in version …” as h3, and the categories as h4?

Comment (friedbunny) [reply platform/ios/scripts/release-notes-jazzy.md.ejs]: > How about “Recent changes” as h2, “Changes in version …” as h3, and the categories as h4?

mapbox maps sdk for ios reference 2

... yeah, “Changes in version” still feels redundant.

Comment (friedbunny) [reply platform/ios/scripts/release-notes-jazzy.md.ejs]: Here’s without, which is my preference:

mapbox maps sdk for ios reference 20180822

/cc @colleenmcginnis

Comment (friedbunny) [reply platform/ios/scripts/release-notes-jazzy.md.ejs]: Went with bare version numbers in 5570fabef938c9bfee8f04e348f5ae4f0e9d5a6c.

**Comment by julianrex** [comment]
Thanks for fixing that up!

**Pull request merged**